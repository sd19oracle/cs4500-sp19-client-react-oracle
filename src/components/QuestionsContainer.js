import React from 'react'
export default props => {
    const tableContainer = props.serviceQuestions
        .map(serviceQuestion =>
            <tr className={serviceQuestion.id + '-tr'} key={serviceQuestion.id} value={serviceQuestion.id}>
                <td onClick={(e) => {
                    props.navToDetail(serviceQuestion.id)
                }}>{serviceQuestion.title}</td>
                <td onClick={(e) => {
                    props.navToDetail(serviceQuestion.id)
                }}>{serviceQuestion.type}</td>
                <td onClick={(e) => {
                    props.navToDetail(serviceQuestion.id)
                }}>{serviceQuestion.choice}</td>

                <td>
                    <button style={{
                        background: "rgb(237,43,63)",
                        color: "white",
                        textAlign: "center",
                        paddingLeft: "13px",
                        paddingRight: "13px",
                        marginLeft: "5px",
                        marginRight: "5px",
                    }}
                        onClick={() => props.remove(serviceQuestion.id)}>Delete
                        </button>
                    <button style={{
                        background: "rgb(83,189,248)",
                        color: "white",
                        textAlign: "center",
                        paddingLeft: "13px",
                        paddingRight: "13px",
                        marginLeft: "5px",
                        marginRight: "5px",
                    }}
                        onClick={() => props.selectQuestion(serviceQuestion.id)}>Select
                        </button>

                </td>

            </tr>
        )
    return tableContainer;
}