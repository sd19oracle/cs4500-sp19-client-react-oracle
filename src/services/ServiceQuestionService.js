export default class ServiceQuestionService {
    static instance = null;
    static hostname = null
    static urlPrefix = null
    static getInstance() {
        if(ServiceQuestionService.instance === null) {
            ServiceQuestionService.instance = new ServiceQuestionService()
        }
        this.hostname = window.location.hostname
        if (this.hostname === "localhost") {
            this.urlPrefix = 'http://localhost:8080/'
        } else {
            this.urlPrefix = 'https://cs4500-sp19-oracle.herokuapp.com/'
        }
        console.log(this.hostname)
        console.log(this.urlPrefix)
        return this.instance
    }
    findServiceQuestionById = id =>
        fetch(ServiceQuestionService.urlPrefix + `api/servicesSpecificQuestions/${id}`)
            .then(response => response.json())
    findAllServiceQuestions = () =>
        fetch(ServiceQuestionService.urlPrefix + "api/servicesSpecificQuestions")
            .then(response => response.json())
}
