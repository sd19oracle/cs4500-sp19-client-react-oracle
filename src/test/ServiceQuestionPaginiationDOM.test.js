import React from 'react'
import NumberButton from '../components/NumberButtons'
import QuestionsContainer from '../components/QuestionsContainer'
import TestRenderer from 'react-test-renderer'
import ServiceQuestionService from '../services/ServiceQuestionService'
const serviceQuestionService = ServiceQuestionService.getInstance();
import '../services/pagination.mock'


// DOM testing and sort of snapshot for rendering correct number of num button for the first page
test('render num buttons correctly for the first page', () => {
    return serviceQuestionService.findAPage(10, 0)
        .then(content => {
            const testRenderer = TestRenderer.create(
                <NumberButton
                    page_num={content.totalPages}
                    current_page={content.pageable.pageNumber + 1}
                />
            )
            let tree = testRenderer.toJSON()
            expect(tree).toMatchSnapshot()

            const testInstance = testRenderer.root

            // since there are 2 total page, so there should be only 2 buttons
            const number_buttons = testInstance.findAllByType("button")
            expect(number_buttons.length).toBe(2);


            const first_button = testInstance.findByProps({ className: '1-button' })
            const second_button = testInstance.findByProps({ className: '2-button' })

            expect(first_button.props.id).toBe(1);
            // fist button should be selected and highlightened 
            expect(first_button.props.style).toEqual({"backgroundColor": "#69adfc", "opacity": 0.8})
            expect(second_button.props.id).toBe(2);
        })
})

// DOM testing and sort of snapshot testing for rendering correct number of num button for the second page
test('render num buttons correctly for the second page', () => {
    return serviceQuestionService.findAPage(10, 1)
        .then(content => {
            const testRenderer = TestRenderer.create(
                <NumberButton
                    page_num={content.totalPages}
                    current_page={content.pageable.pageNumber + 1}
                />
            )
            let tree = testRenderer.toJSON()
            expect(tree).toMatchSnapshot()

            const testInstance = testRenderer.root

            // since there are 2 total page, so there should be only 2 buttons
            const number_buttons = testInstance.findAllByType("button")
            expect(number_buttons.length).toBe(2);

            const first_button = testInstance.findByProps({ className: '1-button' })
            const second_button = testInstance.findByProps({ className: '2-button' })

            expect(first_button.props.id).toBe(1);
            expect(second_button.props.id).toBe(2);
            // the second button should selected and highlightened
            expect(second_button.props.style).toEqual({"backgroundColor": "#69adfc", "opacity": 0.8})
        })
})

// DOM testing and sort of snapshot testing for generating the first page
test("rendering first page correctly with selected page size", () => {
    return serviceQuestionService.findAPage(10, 0)
        .then(content => {
            const testRenderer = TestRenderer.create(
                <QuestionsContainer
                serviceQuestions={content.content}
                />
            )

            let tree = testRenderer.toJSON()
            expect(tree).toMatchSnapshot()

            const testInstance = testRenderer.root
            // total ten questions
            const number_quetions = testInstance.findAllByType("tr")
            expect(number_quetions.length).toBe(10);

            // first questions check
            const first_question = testInstance.findByProps({className:"88-tr"})
            expect(first_question.props.value).toBe(88);
            // last question check
            const last_question = testInstance.findByProps({className:"144-tr"})
            expect(last_question.props.value).toBe(144);
        })
})

// DOM testing and sort of snapshot testing for generating the second page
test("rendering second page correctly with selected page size", () => {
    return serviceQuestionService.findAPage(10, 1)
        .then(content => {
            const testRenderer = TestRenderer.create(
                <QuestionsContainer
                serviceQuestions={content.content}
                />
            )

            let tree = testRenderer.toJSON()
            expect(tree).toMatchSnapshot()

            const testInstance = testRenderer.root
            // total ten questions
            const number_quetions = testInstance.findAllByType("tr")
            expect(number_quetions.length).toBe(1);

            // first questions check
            const first_question = testInstance.findByProps({className:"168-tr"})
            expect(first_question.props.value).toBe(168);
        })
})

// DOM testing and sort of snapshot testing for generating the empty page
test("rendering empty page correctly with selected page size", () => {
    return serviceQuestionService.findAPage(10, 2)
        .then(content => {
            const testRenderer = TestRenderer.create(
                <QuestionsContainer
                serviceQuestions={content.content}
                />
            )

            let tree = testRenderer.toJSON()
            expect(tree).toMatchSnapshot()

            const testInstance = testRenderer.root
            // total ten questions
            const number_quetions = testInstance.findAllByType("tr")
            expect(number_quetions.length).toBe(0);
        })
})