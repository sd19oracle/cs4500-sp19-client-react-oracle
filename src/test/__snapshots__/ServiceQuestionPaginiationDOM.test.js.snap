// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render num buttons correctly for the first page 1`] = `
Array [
  <button
    className="1-button"
    id={1}
    style={
      Object {
        "backgroundColor": "#69adfc",
        "opacity": 0.8,
      }
    }
  >
    1
  </button>,
  <button
    className="2-button"
    id={2}
  >
    2
  </button>,
]
`;

exports[`render num buttons correctly for the second page 1`] = `
Array [
  <button
    className="1-button"
    id={1}
  >
    1
  </button>,
  <button
    className="2-button"
    id={2}
    style={
      Object {
        "backgroundColor": "#69adfc",
        "opacity": 0.8,
      }
    }
  >
    2
  </button>,
]
`;

exports[`rendering empty page correctly with selected page size 1`] = `null`;

exports[`rendering first page correctly with selected page size 1`] = `
Array [
  <tr
    className="88-tr"
    value={88}
  >
    <td
      onClick={[Function]}
    >
      What classes can you teach?
    </td>
    <td
      onClick={[Function]}
    >
      MUTIPLECHOICE
    </td>
    <td
      onClick={[Function]}
    >
      CS,CE,IS,others
    </td>
    <td>
      <button
        onClick={[Function]}
        style={
          Object {
            "background": "rgb(237,43,63)",
            "color": "white",
            "marginLeft": "5px",
            "marginRight": "5px",
            "paddingLeft": "13px",
            "paddingRight": "13px",
            "textAlign": "center",
          }
        }
      >
        Delete
      </button>
      <button
        onClick={[Function]}
        style={
          Object {
            "background": "rgb(83,189,248)",
            "color": "white",
            "marginLeft": "5px",
            "marginRight": "5px",
            "paddingLeft": "13px",
            "paddingRight": "13px",
            "textAlign": "center",
          }
        }
      >
        Select
      </button>
    </td>
  </tr>,
  <tr
    className="123-tr"
    value={123}
  >
    <td
      onClick={[Function]}
    >
      Are you allergic to pollen?
    </td>
    <td
      onClick={[Function]}
    >
      TRUEFALSE
    </td>
    <td
      onClick={[Function]}
    >
      1.no2.yes
    </td>
    <td>
      <button
        onClick={[Function]}
        style={
          Object {
            "background": "rgb(237,43,63)",
            "color": "white",
            "marginLeft": "5px",
            "marginRight": "5px",
            "paddingLeft": "13px",
            "paddingRight": "13px",
            "textAlign": "center",
          }
        }
      >
        Delete
      </button>
      <button
        onClick={[Function]}
        style={
          Object {
            "background": "rgb(83,189,248)",
            "color": "white",
            "marginLeft": "5px",
            "marginRight": "5px",
            "paddingLeft": "13px",
            "paddingRight": "13px",
            "textAlign": "center",
          }
        }
      >
        Select
      </button>
    </td>
  </tr>,
  <tr
    className="124-tr"
    value={124}
  >
    <td
      onClick={[Function]}
    >
      What is your education level?
    </td>
    <td
      onClick={[Function]}
    >
      MUTIPLECHOICE
    </td>
    <td
      onClick={[Function]}
    >
      BS/BA,MS/MA,PhD,others
    </td>
    <td>
      <button
        onClick={[Function]}
        style={
          Object {
            "background": "rgb(237,43,63)",
            "color": "white",
            "marginLeft": "5px",
            "marginRight": "5px",
            "paddingLeft": "13px",
            "paddingRight": "13px",
            "textAlign": "center",
          }
        }
      >
        Delete
      </button>
      <button
        onClick={[Function]}
        style={
          Object {
            "background": "rgb(83,189,248)",
            "color": "white",
            "marginLeft": "5px",
            "marginRight": "5px",
            "paddingLeft": "13px",
            "paddingRight": "13px",
            "textAlign": "center",
          }
        }
      >
        Select
      </button>
    </td>
  </tr>,
  <tr
    className="127-tr"
    value={127}
  >
    <td
      onClick={[Function]}
    >
      What is best way to contact you?
    </td>
    <td
      onClick={[Function]}
    >
      SHORTANSWER
    </td>
    <td
      onClick={[Function]}
    >
      
    </td>
    <td>
      <button
        onClick={[Function]}
        style={
          Object {
            "background": "rgb(237,43,63)",
            "color": "white",
            "marginLeft": "5px",
            "marginRight": "5px",
            "paddingLeft": "13px",
            "paddingRight": "13px",
            "textAlign": "center",
          }
        }
      >
        Delete
      </button>
      <button
        onClick={[Function]}
        style={
          Object {
            "background": "rgb(83,189,248)",
            "color": "white",
            "marginLeft": "5px",
            "marginRight": "5px",
            "paddingLeft": "13px",
            "paddingRight": "13px",
            "textAlign": "center",
          }
        }
      >
        Select
      </button>
    </td>
  </tr>,
  <tr
    className="130-tr"
    value={130}
  >
    <td
      onClick={[Function]}
    >
      test question 3
    </td>
    <td
      onClick={[Function]}
    >
      SHORTANSWER
    </td>
    <td
      onClick={[Function]}
    >
      
    </td>
    <td>
      <button
        onClick={[Function]}
        style={
          Object {
            "background": "rgb(237,43,63)",
            "color": "white",
            "marginLeft": "5px",
            "marginRight": "5px",
            "paddingLeft": "13px",
            "paddingRight": "13px",
            "textAlign": "center",
          }
        }
      >
        Delete
      </button>
      <button
        onClick={[Function]}
        style={
          Object {
            "background": "rgb(83,189,248)",
            "color": "white",
            "marginLeft": "5px",
            "marginRight": "5px",
            "paddingLeft": "13px",
            "paddingRight": "13px",
            "textAlign": "center",
          }
        }
      >
        Select
      </button>
    </td>
  </tr>,
  <tr
    className="136-tr"
    value={136}
  >
    <td
      onClick={[Function]}
    >
      kk
    </td>
    <td
      onClick={[Function]}
    >
      SHORTANSWER
    </td>
    <td
      onClick={[Function]}
    >
      
    </td>
    <td>
      <button
        onClick={[Function]}
        style={
          Object {
            "background": "rgb(237,43,63)",
            "color": "white",
            "marginLeft": "5px",
            "marginRight": "5px",
            "paddingLeft": "13px",
            "paddingRight": "13px",
            "textAlign": "center",
          }
        }
      >
        Delete
      </button>
      <button
        onClick={[Function]}
        style={
          Object {
            "background": "rgb(83,189,248)",
            "color": "white",
            "marginLeft": "5px",
            "marginRight": "5px",
            "paddingLeft": "13px",
            "paddingRight": "13px",
            "textAlign": "center",
          }
        }
      >
        Select
      </button>
    </td>
  </tr>,
  <tr
    className="137-tr"
    value={137}
  >
    <td
      onClick={[Function]}
    >
      Do you have pet(s)?
    </td>
    <td
      onClick={[Function]}
    >
      TRUEFALSE
    </td>
    <td
      onClick={[Function]}
    >
      
    </td>
    <td>
      <button
        onClick={[Function]}
        style={
          Object {
            "background": "rgb(237,43,63)",
            "color": "white",
            "marginLeft": "5px",
            "marginRight": "5px",
            "paddingLeft": "13px",
            "paddingRight": "13px",
            "textAlign": "center",
          }
        }
      >
        Delete
      </button>
      <button
        onClick={[Function]}
        style={
          Object {
            "background": "rgb(83,189,248)",
            "color": "white",
            "marginLeft": "5px",
            "marginRight": "5px",
            "paddingLeft": "13px",
            "paddingRight": "13px",
            "textAlign": "center",
          }
        }
      >
        Select
      </button>
    </td>
  </tr>,
  <tr
    className="139-tr"
    value={139}
  >
    <td
      onClick={[Function]}
    >
      test for fun
    </td>
    <td
      onClick={[Function]}
    >
      MINMAX
    </td>
    <td
      onClick={[Function]}
    >
      
    </td>
    <td>
      <button
        onClick={[Function]}
        style={
          Object {
            "background": "rgb(237,43,63)",
            "color": "white",
            "marginLeft": "5px",
            "marginRight": "5px",
            "paddingLeft": "13px",
            "paddingRight": "13px",
            "textAlign": "center",
          }
        }
      >
        Delete
      </button>
      <button
        onClick={[Function]}
        style={
          Object {
            "background": "rgb(83,189,248)",
            "color": "white",
            "marginLeft": "5px",
            "marginRight": "5px",
            "paddingLeft": "13px",
            "paddingRight": "13px",
            "textAlign": "center",
          }
        }
      >
        Select
      </button>
    </td>
  </tr>,
  <tr
    className="143-tr"
    value={143}
  >
    <td
      onClick={[Function]}
    >
      TESTNOREASON
    </td>
    <td
      onClick={[Function]}
    >
      MUTIPLECHOICE
    </td>
    <td
      onClick={[Function]}
    >
      
    </td>
    <td>
      <button
        onClick={[Function]}
        style={
          Object {
            "background": "rgb(237,43,63)",
            "color": "white",
            "marginLeft": "5px",
            "marginRight": "5px",
            "paddingLeft": "13px",
            "paddingRight": "13px",
            "textAlign": "center",
          }
        }
      >
        Delete
      </button>
      <button
        onClick={[Function]}
        style={
          Object {
            "background": "rgb(83,189,248)",
            "color": "white",
            "marginLeft": "5px",
            "marginRight": "5px",
            "paddingLeft": "13px",
            "paddingRight": "13px",
            "textAlign": "center",
          }
        }
      >
        Select
      </button>
    </td>
  </tr>,
  <tr
    className="144-tr"
    value={144}
  >
    <td
      onClick={[Function]}
    >
      hello
    </td>
    <td
      onClick={[Function]}
    >
      TRUEFALSE
    </td>
    <td
      onClick={[Function]}
    >
      
    </td>
    <td>
      <button
        onClick={[Function]}
        style={
          Object {
            "background": "rgb(237,43,63)",
            "color": "white",
            "marginLeft": "5px",
            "marginRight": "5px",
            "paddingLeft": "13px",
            "paddingRight": "13px",
            "textAlign": "center",
          }
        }
      >
        Delete
      </button>
      <button
        onClick={[Function]}
        style={
          Object {
            "background": "rgb(83,189,248)",
            "color": "white",
            "marginLeft": "5px",
            "marginRight": "5px",
            "paddingLeft": "13px",
            "paddingRight": "13px",
            "textAlign": "center",
          }
        }
      >
        Select
      </button>
    </td>
  </tr>,
]
`;

exports[`rendering second page correctly with selected page size 1`] = `
<tr
  className="168-tr"
  value={168}
>
  <td
    onClick={[Function]}
  >
    ssg
  </td>
  <td
    onClick={[Function]}
  >
    MINMAX
  </td>
  <td
    onClick={[Function]}
  >
    
  </td>
  <td>
    <button
      onClick={[Function]}
      style={
        Object {
          "background": "rgb(237,43,63)",
          "color": "white",
          "marginLeft": "5px",
          "marginRight": "5px",
          "paddingLeft": "13px",
          "paddingRight": "13px",
          "textAlign": "center",
        }
      }
    >
      Delete
    </button>
    <button
      onClick={[Function]}
      style={
        Object {
          "background": "rgb(83,189,248)",
          "color": "white",
          "marginLeft": "5px",
          "marginRight": "5px",
          "paddingLeft": "13px",
          "paddingRight": "13px",
          "textAlign": "center",
        }
      }
    >
      Select
    </button>
  </td>
</tr>
`;
